### PR TITLE
Tnaflix 500 err

### LINKED ISSUE #5107: Tnaflix HTTP Error 500: Internal Server Error
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2022.09.01** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

_No response_

### Provide a description that is worded well enough to be understood

HTTP Error 500: Internal Server Error

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
F:\downloader\tnaflix>yt-dlp https://www.tnaflix.com/hd-videos/Pantat-Bulat-dan-Montok-Dedek-Hijab-bikin-Tegang/video4651785 --verbose
[debug] Command-line config: ['https://www.tnaflix.com/hd-videos/Pantat-Bulat-dan-Montok-Dedek-Hijab-bikin-Tegang/video4651785', '--verbose']
[debug] User config "D:\OS Files\Users\X\AppData\Roaming\yt-dlp\config": ['--continue', '--format-sort', 'vbr,abr', '--http-chunk-size', '1M', '--buffer-size', '16K', '--sub-lang', 'english', '--no-embed-thumbnail', '--embed-metadata', '--abort-on-unavailable-fragment', '-o', '%(uploader)s\\%(title)s [%(id)s].%(ext)s', '-f', 'bestvideo+bestaudio/Source/hls-original/0/4320p/2160p/1440p/1080p/720p/540p/480p/240p', '--ffmpeg-location', 'D:\\OS Files\\Program Files\\ffmpeg\\bin', '--no-lock']
[debug] Encodings: locale cp65001, fs utf-8, pref cp65001, out utf-8, error utf-8, screen utf-8
[debug] ytdl-yt-dlp version 2022.09.01 [5d7c7d656]
[debug] ** Please make bug reports at https://github.com/ytdl-yt-dlp/ytdl-yt-dlp/issues/new/choose instead.
[debug] Lazy loading extractors is disabled
[debug] Python 3.10.4 (CPython 64bit) - Windows-10-10.0.19043-SP0
[debug] Checking exe version: "D:\OS Files\Program Files\ffmpeg\bin\ffmpeg" -bsfs
[debug] Checking exe version: "D:\OS Files\Program Files\ffmpeg\bin\ffprobe" -bsfs
[debug] exe versions: ffmpeg 5.0.1-essentials_build-www.gyan.dev (setts), ffprobe 5.0.1-essentials_build-www.gyan.dev
[debug] Optional libraries: Cryptodome-3.14.1, WebSocketsWrapper-None, brotli-1.0.9, certifi-2021.10.08, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.2
[debug] Proxy map: {}
[debug] Loaded 1717 extractors
[debug] [TNAFlix] Extracting URL: https://www.tnaflix.com/hd-videos/Pantat-Bulat-dan-Montok-Dedek-Hijab-bikin-Tegang/video4651785
[TNAFlix] Pantat-Bulat-dan-Montok-Dedek-Hijab-bikin-Tegang: Downloading webpage
[TNAFlix] [debug] Fetching webpage from https://www.tnaflix.com/hd-videos/Pantat-Bulat-dan-Montok-Dedek-Hijab-bikin-Tegang/video4651785
[TNAFlix] Pantat-Bulat-dan-Montok-Dedek-Hijab-bikin-Tegang: Downloading metadata
[TNAFlix] [debug] Fetching webpage from https://cdn-fck.tnaflix.com/tnaflix/ce9536510b66e11c6bd6.fid?key=e16735fe01baed06dea5fc291891508d&VID=4651785&premium=1&vip=1&alpha
ERROR: [TNAFlix] 4651785: Unable to download XML: HTTP Error 500: Internal Server Error (caused by <HTTPError 500: 'Internal Server Error'>); please report this issue on  https://github.com/ytdl-yt-dlp/ytdl-yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "C:\Python\Python310\lib\site-packages\yt_dlp\extractor\common.py", line 702, in extract
    ie_result = self._real_extract(url)
  File "C:\Python\Python310\lib\site-packages\yt_dlp\extractor\tnaflix.py", line 93, in _real_extract
    cfg_xml = self._download_xml(
  File "C:\Python\Python310\lib\site-packages\yt_dlp\extractor\common.py", line 1087, in download_content
    res = getattr(self, download_handle.__name__)(url_or_request, video_id, **kwargs)
  File "C:\Python\Python310\lib\site-packages\yt_dlp\extractor\common.py", line 1046, in download_handle
    res = self._download_webpage_handle(
  File "C:\Python\Python310\lib\site-packages\yt_dlp\extractor\common.py", line 912, in _download_webpage_handle
    urlh = self._request_webpage(url_or_request, video_id, note, errnote, fatal, data=data, headers=headers, query=query, expected_status=expected_status)
  File "C:\Python\Python310\lib\site-packages\yt_dlp\extractor\common.py", line 857, in _request_webpage
    raise ExtractorError(errmsg, cause=err)

  File "C:\Python\Python310\lib\site-packages\yt_dlp\extractor\common.py", line 839, in _request_webpage
    return self._downloader.urlopen(self._create_request(url_or_request, data, headers, query))
  File "C:\Python\Python310\lib\site-packages\yt_dlp\YoutubeDL.py", line 3079, in wrapper
    result = func(self, *args, **kwargs)
  File "C:\Python\Python310\lib\site-packages\yt_dlp\YoutubeDL.py", line 3979, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "C:\Python\Python310\lib\urllib\request.py", line 525, in open
    response = meth(req, response)
  File "C:\Python\Python310\lib\urllib\request.py", line 634, in http_response
    response = self.parent.error(
  File "C:\Python\Python310\lib\urllib\request.py", line 563, in error
    return self._call_chain(*args)
  File "C:\Python\Python310\lib\urllib\request.py", line 496, in _call_chain
    result = func(*args)
  File "C:\Python\Python310\lib\urllib\request.py", line 643, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 500: Internal Server Error
```

### PR DESCRIPTION
**Pull request by SG5**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

</details>

Fix Tnaflix 500 error code for video download

this MR

```
yt-dlp is up to date (2022.10.04)
[debug] [TNAFlix] Extracting URL: https://www.tnaflix.com/hd-videos/Pantat-Bulat-dan-Montok-Dedek-Hijab-bikin-Tegang/video4651785
[TNAFlix] Pantat-Bulat-dan-Montok-Dedek-Hijab-bikin-Tegang: Downloading webpage
[TNAFlix] Pantat-Bulat-dan-Montok-Dedek-Hijab-bikin-Tegang: Downloading metadata
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: best/bestvideo+bestaudio
[info] 4651785: Downloading 1 format(s): 720p
[download] Pantat Bulat dan Montok Dedek Hijab bikin Tegang [4651785].mp4 has already been downloaded
[download] 100% of   25.23MiB
[debug] Invoking http downloader on "http://fck-ch28.tnaflix.com/dev28/0/030/216/0030216658/Pantat_Bulat_dan_Montok_Dedek_Hijab_bikin_Tegang-720p.mp4?speed=268&burst=1097707&rb=1097707&rs=268&se=1665016237&ss=c133cee0177f0eba15e5f54a81908e8a"
```

current branch

```
yt-dlp is up to date (2022.10.04)
[debug] [TNAFlix] Extracting URL: https://www.tnaflix.com/hd-videos/Pantat-Bulat-dan-Montok-Dedek-Hijab-bikin-Tegang/video4651785
[TNAFlix] Pantat-Bulat-dan-Montok-Dedek-Hijab-bikin-Tegang: Downloading webpage
[TNAFlix] Pantat-Bulat-dan-Montok-Dedek-Hijab-bikin-Tegang: Downloading metadata
ERROR: [TNAFlix] 4651785: Unable to download XML: HTTP Error 500: Internal Server Error (caused by <HTTPError 500: 'Internal Server Error'>); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "C:\Users\admin\Documents\project\yt-dlp\yt_dlp\extractor\common.py", line 672, in extract
    ie_result = self._real_extract(url)
  File "C:\Users\admin\Documents\project\yt-dlp\yt_dlp\extractor\tnaflix.py", line 95, in _real_extract
    transform_source=fix_xml_ampersands, headers={'Referer': url})
  File "C:\Users\admin\Documents\project\yt-dlp\yt_dlp\extractor\common.py", line 1032, in download_content
    res = getattr(self, download_handle.__name__)(url_or_request, video_id, **kwargs)
  File "C:\Users\admin\Documents\project\yt-dlp\yt_dlp\extractor\common.py", line 998, in download_handle
    data=data, headers=headers, query=query, expected_status=expected_status)
  File "C:\Users\admin\Documents\project\yt-dlp\yt_dlp\extractor\common.py", line 866, in _download_webpage_handle
    urlh = self._request_webpage(url_or_request, video_id, note, errnote, fatal, data=data, headers=headers, query=query, expected_status=expected_status)
  File "C:\Users\admin\Documents\project\yt-dlp\yt_dlp\extractor\common.py", line 823, in _request_webpage
    raise ExtractorError(errmsg, cause=err)

  File "C:\Users\admin\Documents\project\yt-dlp\yt_dlp\extractor\common.py", line 805, in _request_webpage
    return self._downloader.urlopen(self._create_request(url_or_request, data, headers, query))
  File "C:\Users\admin\Documents\project\yt-dlp\yt_dlp\YoutubeDL.py", line 3684, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "C:\Python37\lib\urllib\request.py", line 531, in open
    response = meth(req, response)
  File "C:\Python37\lib\urllib\request.py", line 641, in http_response
    'http', request, response, code, msg, hdrs)
  File "C:\Python37\lib\urllib\request.py", line 569, in error
    return self._call_chain(*args)
  File "C:\Python37\lib\urllib\request.py", line 503, in _call_chain
    result = func(*args)
  File "C:\Python37\lib\urllib\request.py", line 649, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 500: Internal Server Error
```

Fixes #5107

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

File: yt_dlp/extractor/tnaflix.py [hunk 1]
```diff
@@ -19,10 +19,15 @@ class TNAFlixNetworkBaseIE(InfoExtractor):
         r'config\s*=\s*(["\'])(?P<url>(?:https?:)?//(?:(?!\1).)+)\1',
     ]
     _HOST = 'tna'
+    _VIDEO_XML_URL = "https://www.tnaflix.com/cdn/cdn.php?file={}.fid&key={}&VID={}&nomp4=1&catID=0&rollover=1&" \
+                     "startThumb=12&embed=0&utm_source=0&multiview=0&premium=1&country=0user=0&vip=1&cd=0&ref=0&alpha"
```
Comment (pukkandan) [inline yt_dlp/extractor/tnaflix.py:23]:
```suggestion
    _VIDEO_XML_URL = 'https://www.tnaflix.com/cdn/cdn.php?file={}.fid&key={}&VID={}&nomp4=1&catID=0&rollover=1&startThumb=12&embed=0&utm_source=0&multiview=0&premium=1&country=0user=0&vip=1&cd=0&ref=0&alpha'
```

File: yt_dlp/extractor/tnaflix.py [hunk 2]
```diff
@@ -85,6 +94,13 @@ def _real_extract(self, url):
             self._CONFIG_REGEX, webpage, 'flashvars.config', default=None,
             group='url'), 'http:')
 
+        if not cfg_url:
+            vkey = extract_field(self._VKEY_REGEX, 'vkey')
+            nkey = extract_field(self._NKEY_REGEX, 'nkey')
+            vid = extract_field(self._VID_REGEX, 'vid')
```
Comment (pukkandan) [inline yt_dlp/extractor/tnaflix.py:100]: put the regexes directly here

File: yt_dlp/extractor/tnaflix.py [hunk 3]
```diff
@@ -71,6 +76,10 @@ def get_child(elem, names):
     def _real_extract(self, url):
         mobj = self._match_valid_url(url)
         video_id = mobj.group('id')
+
+        def extract_field(pattern, name):
+            return self._html_search_regex(pattern, webpage, name, default=None) if pattern else None
```
Comment (pukkandan) [inline yt_dlp/extractor/tnaflix.py:81]: This seems unnecessary. But I'll let it be since it's legacy code

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/tnaflix.py (hunk 1 above)
Comment (SG5) [reply yt_dlp/extractor/tnaflix.py:23]: Thanks, I saw 120 characters, but missed it's a soft limit

File: yt_dlp/extractor/tnaflix.py (hunk 3 above)
Comment (SG5) [reply yt_dlp/extractor/tnaflix.py:81]: Decided to use same way to extract data as it's right now in this file

File: yt_dlp/extractor/tnaflix.py [hunk 4]
```diff
@@ -85,6 +90,13 @@ def _real_extract(self, url):
             self._CONFIG_REGEX, webpage, 'flashvars.config', default=None,
             group='url'), 'http:')
 
+        if not cfg_url:
+            vkey = extract_field(r'<input id="vkey" type="hidden" value="(?P<vkey>[^"]+)" />', 'vkey')
+            nkey = extract_field(r'<input id="nkey" type="hidden" value="(?P<nkey>[^"]+)" />', 'nkey')
+            vid = extract_field(r'<input id="VID" type="hidden" value="(?P<VID>[^"]+)" />', 'vid')
```
Comment (pukkandan) [inline yt_dlp/extractor/tnaflix.py:96]:
```suggestion
            vkey = extract_field(r'<input id="vkey" type="hidden" value="([^"]+)"', 'vkey')
            nkey = extract_field(r'<input id="nkey" type="hidden" value="([^"]+)"', 'nkey')
            vid = extract_field(r'<input id="VID" type="hidden" value="([^"]+)"', 'vid')
```

File: yt_dlp/extractor/tnaflix.py [hunk 5]
```diff
@@ -85,6 +90,13 @@ def _real_extract(self, url):
             self._CONFIG_REGEX, webpage, 'flashvars.config', default=None,
             group='url'), 'http:')
 
+        if not cfg_url:
+            vkey = extract_field(r'<input id="vkey" type="hidden" value="([^"]+)"', 'vkey')
+            nkey = extract_field(r'<input id="nkey" type="hidden" value="([^"]+)"', 'nkey')
+            vid = extract_field(r'<input id="VID" type="hidden" value="([^"]+)"', 'vid')
```
Comment (pukkandan) [inline yt_dlp/extractor/tnaflix.py:96]:
```suggestion
            vkey = extract_field(r'<input\b[^>]+\bid="vkey"\b[^>]+\bvalue="([^"]+)"', 'vkey')
            nkey = extract_field(r'<input\b[^>]+\bid="nkey"\b[^>]+\bvalue="([^"]+)"', 'nkey')
            vid = extract_field(r'<input\b[^>]+\bid="VID"\b[^>]+\bvalue="([^"]+)"', 'vid')
```

**Comment by pukkandan** [comment]
Pls test my changes and let me know

**Comment by SG5** [comment]
With changes everything works fine

```
yt-dlp is up to date (2022.10.04)
[TNAFlix] Pantat-Bulat-dan-Montok-Dedek-Hijab-bikin-Tegang: Downloading webpage
[debug] [TNAFlix] Extracting URL: https://www.tnaflix.com/hd-videos/Pantat-Bulat-dan-Montok-Dedek-Hijab-bikin-Tegang/video4651785
[TNAFlix] Pantat-Bulat-dan-Montok-Dedek-Hijab-bikin-Tegang: Downloading metadata
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: best/bestvideo+bestaudio
[info] 4651785: Downloading 1 format(s): 720p
[download] Pantat Bulat dan Montok Dedek Hijab bikin Tegang [4651785].mp4 has already been downloaded
[download] 100% of   25.23MiB
[debug] Invoking http downloader on "http://fck-cl28.tnaflix.com/dev28/0/030/216/0030216658/Pantat_Bulat_dan_Montok_Dedek_Hijab_bikin_Tegang-720p.mp4?speed=274432&end=1665035319&burst=1097707&secure=03509372ccb35d23164ea"
```

```
yt-dlp is up to date (2022.10.04)
[TNAFlix] Fiona-Lewis-Breasts-Scene-in-Tintorera: Downloading webpage
[debug] [TNAFlix] Extracting URL: https://www.tnaflix.com/amateur-porn/Fiona-Lewis-Breasts-Scene-in-Tintorera/video4674185?isFeatured=1
[TNAFlix] Fiona-Lewis-Breasts-Scene-in-Tintorera: Downloading metadata
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: best/bestvideo+bestaudio
[info] 4674185: Downloading 1 format(s): 360p
[debug] Invoking http downloader on "http://fck-cl38.tnaflix.com/dev38/0/030/294/0030294007/Fiona_Lewis_Breasts_Scene_in_Tintorera-360p.mp4?speed=93184&end=1665035433&burst=294461&secure=0de578bf4f6f73028847c"
[debug] File locking is not supported. Proceeding without locking
[download] Destination: Fiona Lewis Breasts Scene  in Tintorera [4674185].mp4
[download] 100% of    1.46MiB in 00:00:12 at 118.11KiB/s 

Process finished with exit code 0
```

**Pull request merged**